### PR TITLE
[KED-2970] Improve micropackaging CLI experience

### PR DESCRIPTION
**Pull request by SajidAlamQB**
## Description

Current problems: * The kedro pipeline command group from 0.18.0 will also package things that are not pipelines and this current naming will lead users to think the kedro pipeline package and kedro pipeline pull commands only work on pipelines

The kedro pipeline command group from 0.18.0 also has commands that don't work on things that are not pipeline modules e.g. kedro pipeline delete only works on pipelines
 
Solution: * Create a new command group called `kedro micropkg` to capture being able to package and pull things that are not pipelines.

@yetudada agreed to go with `kedro micropkg` as the new command group on the cli. `kedro micropkg` will live alongside `kedro pipeline pull/package` for 0.17.7 with deprecation warnings but will be removed in 0.18.0.

related to: https://github.com/kedro-org/kedro/pull/1194

https://jira.quantumblack.com/browse/KED-2970

## Development notes

Documentation and refactoring code will be done in a separate PR.

Imported several methods from `pipeline.py` to `micropkg.py` including `_install_files`, `_generate_wheel_file` etc..

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1224"></a>

### CODE REVIEW

File: tests/framework/cli/micropkg/test_micropkg_package.py
Comment (lorenabalan) [inline tests/framework/cli/micropkg/test_micropkg_package.py:438]: We need to mention this in the release notes as well.

File: tests/framework/cli/micropkg/test_micropkg_pull.py
Comment (lorenabalan) [inline tests/framework/cli/micropkg/test_micropkg_pull.py:237]: as mentioned before, these will turn into `sdist_file = ...` or something of the sort.

**Review by lorenabalan — COMMENTED** [review]
LGTM, happy to approve when all the tests builds pass. Don't forget to add this in the release notes, as it directly affects the user. We need to also update documentation to make sure this cli group is included and that our tutorials refer to this one instead of `pipeline`. But I think that should be a separate PR, to make it easier to review. You can make it branch off this branch, or off main directly if you prefer.

I really appreciate that you copy-pasted only and did minimal changes to make it work for the new name - thus not mixing refactoring and copied changes in one go, which would make it hard to review. In a separate PR we could refactor some of those variable names and functions so that they don't talk about `pipelines` but more generally about `packages`, `micropackages` etc. But maybe we can do that directly on `develop` and we live with it on `main`. This is not a huge deal, but it does keep our code base in a healthy state if we do it.

File: kedro/framework/cli/micropkg.py
Comment (MerelTheisenQB) [inline kedro/framework/cli/micropkg.py:111]: This command is now not only for `pipelines` anymore right? Since this help message is user facing, I think we should update it to reflect the functionality. And the same with the help messages for the other CLI commands in this file.

Comment (SajidAlamQB) [reply kedro/framework/cli/micropkg.py:111]: Agreed I will update all help messages to reflect this.

File: kedro/framework/cli/micropkg.py
Comment (idanov) [inline kedro/framework/cli/micropkg.py:251]: Since this command is new, we can directly make the default to be `dist/` rather than `src/dist/` (no need to keep it as it is for backwards compatibility).

**Review by idanov — APPROVED** [review]

File: kedro/framework/cli/micropkg.py
Comment (MerelTheisenQB) [inline kedro/framework/cli/micropkg.py:92]: This one should be pull all `micro-packages` as well right?

File: kedro/framework/cli/micropkg.py
Comment (MerelTheisenQB) [inline kedro/framework/cli/micropkg.py:89]: This isn't user facing, but let's update this as well to modular pipeline and other micro-packages.

File: kedro/framework/cli/micropkg.py
Comment (MerelTheisenQB) [inline kedro/framework/cli/micropkg.py:103]: Another one to update

File: kedro/framework/cli/micropkg.py
Comment (MerelTheisenQB) [inline kedro/framework/cli/micropkg.py:163]: `Micro-packages` pulled & unpacked

File: kedro/framework/cli/micropkg.py
Comment (MerelTheisenQB) [inline kedro/framework/cli/micropkg.py:187]: Update to micro-package.

File: kedro/framework/cli/micropkg.py
Comment (MerelTheisenQB) [inline kedro/framework/cli/micropkg.py:231]: Let's update this command as well, because it's also user facing.

File: kedro/framework/cli/micropkg.py
Comment (MerelTheisenQB) [inline kedro/framework/cli/micropkg.py:244]: Same here.

**Review by MerelTheisenQB — COMMENTED** [review]
I think this is nearly done! 👍  Just some `click` messages that need to be updated, because those are user facing as well.

File: kedro/framework/cli/micropkg.py
Comment (MerelTheisenQB) [reply kedro/framework/cli/micropkg.py:205]: I think this still needs to be updated right?

**Review by MerelTheisenQB — APPROVED** [review]
Great work @SajidAlamQB 👏 ⭐

File: RELEASE.md
Comment (lorenabalan) [inline RELEASE.md:7]: This last one doesn't look right. Functionality was already there. I meant the migration notes should mention [tool.blabla.micropkg] instead of [tool.blabla.pipeline], but I guess that's for `develop` PR.

File: RELEASE.md
Comment (lorenabalan) [inline RELEASE.md:21]: We're still writing to humans, not robots. 😂 
```suggestion
* `kedro pipeline pull` and `kedro pipeline package` will be deprecated. Please use `kedro micropkg` instead.
```

File: tests/framework/cli/micropkg/test_micropkg_requirements.py
Comment (lorenabalan) [inline tests/framework/cli/micropkg/test_micropkg_requirements.py:53]: We can maybe refactor to rename this `call_micropkg_package`.

File: tests/framework/cli/micropkg/test_micropkg_requirements.py
Comment (lorenabalan) [inline tests/framework/cli/micropkg/test_micropkg_requirements.py:67]: Similarly here, `call_micropkg_pull`.

File: kedro/framework/cli/micropkg.py
Comment (lorenabalan) [inline kedro/framework/cli/micropkg.py:278]: Worth mentioning in the release notes that the difference between `micropkg` and `pipeline` is that the former saves packages to `project / dist ` and the latter in `project / src / dist`, so that users are aware that they are not 100% equivalent.

**Review by lorenabalan — CHANGES_REQUESTED** [review]
Almost there - please delete `test_micropkg.py` first. I would also hold off merging until https://github.com/kedro-org/kedro/pull/1235 is merged and the one raised after it is merged as well (because there's quite a few commits that haven't made it to develop yet, and the conflicts will be a nightmare to sort out). If we can keep this commit in its separate AUTO-MERGE pr that'd be great so it's easier to follow the changes.

**Review by lorenabalan — APPROVED** [review]
Feel free to merge this, all commits on `main` should've reached `develop` as well. :D

**Pull request merged**
